src/plugins/intel_gpu/src/kernel_selector/jitter.cpp

```diff
@@ -305,70 +310,83 @@ class TensorBaseTJitConstant : public JitConstant {
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 class DataTensorJitConstant : public TensorBaseTJitConstant<Datatype, DataLayout> {
     const DataTensor _tensor;
-    const size_t _dyn_array_index;
 
 public:
-    DataTensorJitConstant(const std::string& name, const DataTensor& t, size_t dyn_array_index = 0)
+    DataTensorJitConstant(const std::string& name, const DataTensor& t)
     : TensorBaseTJitConstant(name)
-    , _tensor(t)
-    , _dyn_array_index(dyn_array_index) {}
+    , _tensor(t) {}
 
     JitDefinitions GetDefinitions() const override;
 };
 
 JitDefinitions DataTensorJitConstant::GetDefinitions() const {
     JitDefinitions baseDefinitions = TensorBaseTJitConstant::GetDefinitions(_tensor);
-    JitDefinitions definitions{};
 
-    DimensionAccessHelper dims(_tensor, _dyn_array_index);
-    DimensionAccessHelper dims_padded(_tensor, _dyn_array_index, true);
+    JitDefinitions definitions{};
+    DimensionAccessHelper dims(_tensor);
+    DimensionAccessHelper dims_padded(_tensor, true);
+    // shape_info layout
+    // if only y has dynamic padding:
+    // [dim_b, dim_f, dim_w, dim_z, dim_y, dim_x, pad_before_y, pad_after_y]
+    // if only x has dynamic padding:
+    // [dim_b, dim_f, dim_w, dim_z, dim_y, dim_x, pad_before_x, pad_after_x]
```

Comment:
nit: could you also add dim_v, dim_u in comments to match corresponding 8D update?